Pool_Properties_Design.md

```diff
@@ -0,0 +1,173 @@
+	// IOConnectivity represents some suggested IO connectivity capabilities.
+	IOConnectivity IOConnectivityLoS `json:"ioConnectivity,omitempty" yaml:"ioConnectivity,omitempty"`
+	// DataProtection represents some suggested data protection capabilities.
+	DataProtection DataProtectionLos `json:"dataProtection,omitempty" yaml:"dataProtection,omitempty"`
+
+	// Besides those basic suggested pool properties above, vendors can configure
+	// some advanced features (diskType, IOPS, throughout, latency, etc)
+	// themselves, all these properties can be exposed to controller scheduler
+	// and filtered by selector in a extensible way.
+	Advanced map[string]interface{} `json:"advanced,omitempty" yaml:"advanced,omitempty"`
+}
+```
+
+### REST API impact
+
+None
+
+### Security impact
+
+None
+
+### Other end user impact
+
+None
+
+### Performance impact
+
+Compared with before, there would be some performance loss when encoding and decoding the StoragePoolSpec. But we can improve the performance like this:
+```go
+type StoragePoolExtraSpec struct {
+	// DataStorage represents suggested some data storage capabilities.
+	DataStorage *DataStorageLoS `json:"dataStorage,omitempty" yaml:"dataStorage,omitempty"`
+	// IOConnectivity represents some suggested IO connectivity capabilities.
+	IOConnectivity *IOConnectivityLoS `json:"ioConnectivity,omitempty" yaml:"ioConnectivity,omitempty"`
+	// DataProtection represents some suggested data protection capabilities.
+	DataProtection *DataProtectionLos `json:"dataProtection,omitempty" yaml:"dataProtection,omitempty"`
+
+	// Besides those basic suggested pool properties above, vendors can configure
+	// some advanced features (diskType, IOPS, throughout, latency, etc)
+	// themselves, all these properties can be exposed to controller scheduler
+	// and filtered by selector in a extensible way.
+	Advanced *map[string]interface{} `json:"advanced,omitempty" yaml:"advanced,omitempty"`
+}
+```
+
+### Other deployer impact
+
+If this prososal gets permitted, the backend configuration file (take ceph for example) should be updated like this:
+```yaml
+configFile: /etc/ceph/ceph.conf
+pool:
+  rbd:
+    storageType: block
+    availabilityZone: default
+    extras:
+      dataStorage:
+        provisioningPolicy: Thin
+        isSpaceEfficient: true
+      ioConnectivity:
+        accessProtocol: rbd
+        maxIOPS: 1000
+      advanced:
+        diskType: SSD
+        throughput: 1000
+```
+
+### Developer impact
+
+Because this proposal has modified the `StoragePool` definition, all packages (db, api and so on) could be affected.
+
+## Use Cases
+
+* Multi-tenants resource provisioning
+
+After the system initiated, the capabilities report thread will be triggered automatically and then capabilities properties of storage pool resource will be stored in database. After that, admin A creates a `ProvisionProfile` which can be viewed by all users, so user B can check all available profiles and choose one to create a volume.
```

Comment:
Will be good to provide an example of how the ProvisionProfile looks like.

Reply:
I agree, some description would be good.